src/main/java/uk/gov/hmcts/reform/translate/service/DictionaryService.java

```diff
@@ -59,15 +70,55 @@ public Map<String, String> getDictionaryContents() {
     }
 
 
-    public void putDictionary(final Dictionary dictionaryRequest) {
+    public void putDictionary(Dictionary dictionaryRequest) {
 
         val isManageTranslationRole = securityUtils.hasRole(MANAGE_TRANSLATIONS_ROLE);
+        validateDictionary(dictionaryRequest, isManageTranslationRole);
         val currentUserId = securityUtils.getUserInfo().getUid();
+
         dictionaryRequest.getTranslations().entrySet()
             .stream()
             .forEach(phrase -> processPhrase(phrase, currentUserId, isManageTranslationRole));
     }
 
+    public void putDictionaryRoleCheck(String clientS2SToken) {
+        val clientServiceName = securityUtils.getServiceNameFromS2SToken(clientS2SToken);
+        validateServiceRequest(clientServiceName);
+        isBypassRoleAuthCheck(clientServiceName);
+    }
+
+    private void validateServiceRequest(String clientServiceName) {
+        if (isBypassRoleAuthCheck(clientServiceName)
+            || securityUtils.hasAnyOfThisRoles(Arrays.asList(MANAGE_TRANSLATIONS_ROLE, LOAD_TRANSLATIONS_ROLE))) {
+            return;
+        }
+        throw new RequestErrorException(MANAGE_TRANSLATIONS_ROLE + "," + LOAD_TRANSLATIONS_ROLE);
+    }
+
+    private boolean isBypassRoleAuthCheck(String clientServiceName) {
+        return applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck().contains(clientServiceName);
+    }
+
+    private void validateDictionary(final Dictionary dictionaryRequest, boolean isManageTranslationRole) {
+
+        if (isTranslationEmpty(dictionaryRequest)) {
+            throw new BadRequestException("The translations field cannot be empty.");
```

Comment:
Can we move error message to constant for use in Swagger and Unit/Integration tests.

Reply:
Resolved in 120fa6587700c9064d3a326684db2c0938fc4d33